service/internal/fanoutconsumer/logs.go

```diff
@@ -0,0 +1,78 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package fanoutconsumer contains implementations of Traces/Metrics/Logs consumers
+// that fan out the data to multiple other consumers.
+package fanoutconsumer
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+// NewLogs wraps multiple log consumers in a single one.
+// It fanouts the incoming data to all the consumers, and does smart routing:
+//  * Clones only to the consumer that needs to mutate the data.
+//  * If all consumers needs to mutate the data one will get the original data.
+func NewLogs(lcs []consumer.Logs) consumer.Logs {
```

Comment:
Is this a new function or it existed? It looks like it existed but was moved to a new file which makes very difficult to understand what has changed. Can you split this PR into 2 parts: 1) split files, no logic change. 2) logic changes. Otherwise it is difficult to review.

Reply:
The functionality is completely re-written, can do what you suggest, but would not help.